packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator.Input/src/InputTypes/Serialization/TypeSpecInputOAuth2AuthConverter.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.TypeSpec.Generator.Input
+{
+    internal class TypeSpecInputOAuth2AuthConverter : JsonConverter<InputOAuth2Auth>
```

Comment:
Is this needed just because of the casing changes?

Reply:
yes.
some Json property attribute could also do the trick but it would not handle references (which might work for this type because there would not be references for this)